In [47]:
! pip install yfinance 

# The Spot Rate Cross Exchange Rates and Percentage Changes

In [48]:
import yfinance as yf
import pandas as pd 
import numpy as np
from datetime import datetime, timedelta

In [49]:
Ticker = 'USDEUR=X'
start_date = '2023-01-01'
end_date= '2023-12-31'

In [50]:
data = yf.download(Ticker, start=start_date, end=end_date)

[*********************100%%**********************]  1 of 1 completed


In [51]:
data

Open     High      Low    Close  Adj Close  Volume
Date                                                             
2023-01-02  0.93373  0.93868  0.93350  0.93373    0.93373       0
2023-01-03  0.93653  0.95043  0.93610  0.93653    0.93653       0
2023-01-04  0.94815  0.94823  0.94060  0.94815    0.94815       0
2023-01-05  0.94283  0.95097  0.94050  0.94283    0.94283       0
2023-01-06  0.95037  0.95372  0.94142  0.95037    0.95037       0
...             ...      ...      ...      ...        ...     ...
2023-10-24  0.93722  0.94480  0.93541  0.93722    0.93722       0
2023-10-25  0.94381  0.94636  0.94280  0.94381    0.94381       0
2023-10-26  0.94632  0.95000  0.94610  0.94632    0.94632       0
2023-10-27  0.94657  0.94910  0.94360  0.94657    0.94657       0
2023-10-30  0.94580  0.94780  0.94090  0.94170    0.94170       0

[216 rows x 6 columns]

In [52]:
end_date_0 = datetime.now()
start_date_0 = end_date_0 - timedelta(days=365)

In [53]:
exchange_rate_date_0 = yf.download(Ticker, start=start_date_0, end=end_date_0)

[*********************100%%**********************]  1 of 1 completed


In [54]:
spot_exchange_rate_0 = exchange_rate_date_0['Adj Close'].iloc[-1]

In [55]:
spot_exchange_rate_0

0.9416999816894531

In [56]:
# Define nominal interest rates for the home (EUR) and foreign (USD) countries
nominal_interest_rate_home_0 = 0.035 # Example: 1% nominal interest rate in EUR
nominal_interest_rate_foreign_0 = 0.052 # Example: 2% nominal interest rate in USD


In [57]:
# Calculate the expected future exchange rate (EUR/USD) using the IFE formula
current_exchange_rate = exchange_rate_date_0['Adj Close'].iloc[0]  # Use the first day's rate
expected_exchange_rate = current_exchange_rate * (1 + nominal_interest_rate_foreign_0) / (1 + nominal_interest_rate_home_0)


In [58]:
print(f"Expected future USD/EUR exchange rate: {expected_exchange_rate:.4f}")

Expected future USD/EUR exchange rate: 1.0210


In [59]:
Ticker_1 = 'CHFUSD=X'
start_date_1 = '2023-01-01'
end_date_1 ='2023-12-31'

In [60]:
data_1 = yf.download(Ticker_1, start=start_date_1, end=end_date_1)

[*********************100%%**********************]  1 of 1 completed


In [61]:
data_1.tail()

Open      High       Low     Close  Adj Close  Volume
Date                                                                 
2023-10-24  1.122838  1.125075  1.117568  1.122838   1.122838       0
2023-10-25  1.119746  1.121202  1.115573  1.119746   1.119746       0
2023-10-26  1.115200  1.115573  1.111198  1.115200   1.115200       0
2023-10-27  1.112557  1.113586  1.106929  1.112557   1.112557       0
2023-10-30  1.109139  1.111173  1.105143  1.109152   1.109152       0

In [62]:
end_date_1 = datetime.now()
start_date_1 = end_date_1 - timedelta(days=365)

In [63]:
exchange_rate_date_1 = yf.download(Ticker_1, start=start_date_1, end=end_date_1)

[*********************100%%**********************]  1 of 1 completed


In [64]:
spot_exchange_rate_1 = exchange_rate_date_1['Adj Close'].iloc[-1]

In [65]:
spot_exchange_rate_1

1.109004020690918

In [66]:
# Define nominal interest rates for the home (EUR) and foreign (USD) countries
nominal_interest_rate_home_1 = 0.017  # Example: 1% nominal interest rate in CHF
nominal_interest_rate_foreign_1 = 0.052  # Example: 2% nominal interest rate in USD

In [67]:
# Calculate the expected future exchange rate (EUR/USD) using the IFE formula
current_exchange_rate_1 = exchange_rate_date_1['Adj Close'].iloc[0]  # Use the first day's rate
expected_exchange_rate_1 = current_exchange_rate * (1 + nominal_interest_rate_home_1) / (1 + nominal_interest_rate_foreign_1)

In [68]:
print(f"Expected future CHF/EUR exchange rate: {expected_exchange_rate_1:.4f}")

Expected future CHF/EUR exchange rate: 0.9711


In [69]:
Ticker_2 = 'USDGBP=X'
start_date_2 = '2023-01-01'
end_date_2 = '2023-12-31'

In [70]:
data_2 = yf.download(Ticker_2, start=start_date_2, end=end_date_2)

[*********************100%%**********************]  1 of 1 completed


In [71]:
data_2.tail()

Open     High      Low    Close  Adj Close  Volume
Date                                                             
2023-10-24  0.81623  0.82240  0.81378  0.81631    0.81631       0
2023-10-25  0.82215  0.82538  0.82120  0.82224    0.82224       0
2023-10-26  0.82630  0.82847  0.82427  0.82621    0.82621       0
2023-10-27  0.82399  0.82600  0.82225  0.82392    0.82392       0
2023-10-30  0.82482  0.82698  0.82141  0.82187    0.82187       0

In [72]:
end_date_2 = datetime.now()
start_date_2 = end_date_2 - timedelta(days=365)

In [73]:
exchange_rate_date_2 = yf.download(Ticker_2, start=start_date_2, end=end_date_2)

[*********************100%%**********************]  1 of 1 completed


In [74]:
spot_exchange_rate_2 = exchange_rate_date_2['Adj Close'].iloc[-1]

In [75]:
spot_exchange_rate_2

0.8218700289726257

In [76]:
# Define nominal interest rates for the home (EUR) and foreign (USD) countries
nominal_interest_rate_home_2 = 0.052  # Example: 1% nominal interest rate in GBP
nominal_interest_rate_foreign_2 = 0.052  # Example: 2% nominal interest rate in USD

In [77]:
# Calculate the expected future exchange rate (EUR/USD) using the IFE formula
current_exchange_rate_2 = exchange_rate_date_2['Adj Close'].iloc[0]  # Use the first day's rate
expected_exchange_rate_2 = current_exchange_rate * (1 + nominal_interest_rate_home_2) / (1 + nominal_interest_rate_foreign_2)

In [79]:
print(f"Expected future USD/GBP exchange rate: {expected_exchange_rate_2:.4f}")

Expected future USD/GBP exchange rate: 1.0045


The spot rate CHF/EUR cross-rate is closest to USD/EUR * CHF/USD = 0.9416 * 1.1090 = 1.0442

 The euro is the base currency in the USD/EUR quote, and the 
expected decrease in the USD/EUR rate indicates that the EUR is depreciating 
(in one year it will cost less USD to buy one EUR). = 1.0210/0.9416 -1 = 0.0843